src/test/java/com/squareup/kotlinpoet/AnnotatedTypeNameTest.kt

```diff
@@ -33,52 +33,52 @@ class AnnotatedTypeNameTest {
     val simpleString = String::class.asTypeName()
     assertFalse(simpleString.isAnnotated)
     assertEquals(simpleString, String::class.asTypeName())
-    val annotated = simpleString.annotated(NEVER_NULL)
+    val annotated = simpleString.copy(annotations = simpleString.annotations + NEVER_NULL)
     assertTrue(annotated.isAnnotated)
-    assertEquals(annotated, annotated.annotated())
   }
 
   @Test fun annotatedType() {
     val expected = "@$NN kotlin.String"
     val type = String::class.asTypeName()
-    val actual = type.annotated(NEVER_NULL).toString()
+    val actual = type.copy(annotations = type.annotations + NEVER_NULL).toString()
     assertEquals(expected, actual)
   }
 
   @Test fun annotatedTwice() {
     val expected = "@$NN @java.lang.Override kotlin.String"
     val type = String::class.asTypeName()
-    val actual = type.annotated(NEVER_NULL)
-        .annotated(AnnotationSpec.builder(Override::class).build())
+    val actual = type
+        .copy(annotations = type.annotations + NEVER_NULL +
+            AnnotationSpec.builder(Override::class).build())
```

Comment:
for later, should we have something like `fun Class<out Annotation>.asAnnotationSpec()` ?

Reply:
Yeah, we should